css-scroll-snap-2/Overview.bs

```diff
@@ -275,119 +275,248 @@ Snap Events {#snap-events}
+		1. Fire a {{SnapEvent}}, |snapevent|, named {{snapchanged}} at <var>target</var>
+			and let |snapevent|'s {{SnapEvent/snapTargetBlock}} and
+			{{SnapEvent/snapTargetInline}} attributes be the <a>snapchangedTargetBlock</a> and the
+			<a>snapchangedTargetInline</a>, respectively, that are associated with <var>target</var>.
+	1. Empty <var>doc</var>'s <a>pending snapchanged event targets</a>.
+
+	<h4 id="snapchanging"> snapchanging </h4>
+	{{snapchanging}} is dispatched:
+	* during a scrolling operation, if the element to which a
+		 <a spec=css-scroll-snap lt="scroll snap container">snap container</a> would
+		 <a spec="css-scroll-snap-1" lt="scroll snap">snap</a> (in either axis) changes, or
+	* if a [[css-scroll-snap-1#re-snap|layout change]] occurs such that a {{snapchanged}} event
+		 is to be dispatched. In this case, as with the scrolling case, the {{snapchanging}} event
+		 should be dispatched before the {{snapchanged}} event.
+
+	A scrolling operation may animate towards a particular position (e.g.
+	scrollbar arrow clicks, arrow key presses, "behavior: smooth" programmatic
+	scrolls) or may directly track a user's input (e.g. touch scrolling, scrollbar
+	dragging). In either case, the user agent [[css-scroll-snap-1#choosing|chooses]] an
+	<dfn>eventual snap target</dfn> in each axis to which the scroller will
+	<a spec="css-scroll-snap-1" lt="scroll snap">snap</a> after the scrolling operation
+	reaches its intended scroll position.
+	* In the former case, the intended scroll position is the scroll animation's
+		 target scroll offset.
+	* In the latter case, the intended scroll position is the current scroll offset as
+		 determined by the user's input.
+
+	{{snapchanging}} aims to let the web page know, as early as possible,
+	that the scrolling operation will result in a change in the element the snap
+	container is snapped to. The user agent should evaluate whether to trigger
+	{{snapchanging}} based on the	<a>eventual snap target</a> to which the scroller would
+	<a spec="css-scroll-snap-1" lt="scroll snap">snap</a> were the scrolling operation
+	to reach its intended scroll position.
+
+	Note: Since snapchanging gives the web page hints about future snapping,
+	the snapping hinted at by a snapchanging event may not materialize since it
+	will be	possible for subsequent scrolling input to further alter the snap
+	container's scroll position and result in a different eventual snap target.
+
+
+	{{snapchanging}} events should fire before {{scroll}} events.
+
+	Each {{Document}} has an associated list of
+	<dfn export for=Document>pending snapchanging event targets</dfn>, initially empty.
+
+	Each
+	<a spec=css-scroll-snap lt="scroll snap container">snap container</a> has
+	one <dfn>snapchangingTargetBlock</dfn> (which is either null or an {{Element}})
+	and one <dfn>snapchangingTargetInline</dfn> (which is either null or an
+	{{Element}}) in the block and inline axes respectively.
```

Comment:
nit: I think you only need to say this once. E.g.

	Each
	<a spec=css-scroll-snap lt="scroll snap container">snap container</a> has
	a <dfn>snapchangingTargetBlock</dfn> a <dfn>snapchangingTargetInline</dfn>
	for the block and inline axes respectively which can either be null if not currently snapping in that axis, or an
	{{Element}} to which the scroller is snapping.

Reply:
Thanks!